app/assets/javascripts/discourse/app/services/sidebar-state.js

```diff
@@ -19,12 +21,34 @@ export default class SidebarState extends Service {
   @tracked displaySwitchPanelButtons = false;
   @tracked filter = "";
   previousState = {};
+  #hiders = new Set();
 
   constructor() {
     super(...arguments);
     this.#reset();
   }
 
+  get applicationController() {
+    return getOwnerWithFallback(this).lookup("controller:application");
+  }
+
+  registerHider(ref) {
+    this.#hiders.add(ref);
+    this.applicationController.setProperties({
+      sidebarDisabledRouteOverride: true,
+    });
+
+    registerDestructor(ref, () => {
+      this.#hiders.delete(ref);
+
+      if (this.counter.size === 0) {
+        this.applicationController.setProperties({
+          sidebarDisabledRouteOverride: false,
+        });
+      }
+    });
+  }
```

Comment:
Would be nice to make this more declarative, and remove the 'code smell' of a controller injection. Maybe something like


```js
#hiders = new TrackedSet();

get sidebarHidden(){
  return this.#hiders.size() > 0;
}

registerHider(ref){
  this.#hiders.add(ref);
  registerDestructor(ref, () => {
    this.#hiders.delete(ref);
  }
}
```

And then in applicationController

```js
get sidebarDisabledRouteOverride(){ // or maybe change the name?
  return this.sidebarState.sidebarHidden;
}
```